servers/src/main/java/tachyon/worker/block/TieredBlockStore.java

```diff
@@ -126,73 +135,93 @@ public void unlockBlock(long userId, long blockId) throws NotFoundException {
   @Override
   public BlockWriter getBlockWriter(long userId, long blockId) throws NotFoundException,
       IOException {
-    TempBlockMeta tempBlockMeta = mMetaManager.getTempBlockMeta(blockId);
-    return new LocalFileBlockWriter(tempBlockMeta);
+    // NOTE, a temp block is supposed to be visible for one writer
```

Comment:
for "only" one writer? Is this the reason that the block lock does not need to be acquired? That should be explicit in the comment.